appcelerator/titanium_mobile
# [TIMOB-24707] Android: Show images and videos in picker

**JIRA:** https://jira.appcelerator.org/browse/TIMOB-24707

**Summary:**
Added `MEDIA_TYPE_VIDEO` support to `Ti.Media.openPhotoGallery()` on Android. Can now select both images and videos with this API, like iOS.
_(Does not support live photos. That is an iOS only feature.)_

**Test 1:**
Re-run test attached to PR #9918.

**Test 2:**
Re-run test attached to PR #9906.
Verify that selecting an image from Google Drive (via the cloud) does not cause a crash.

**Test 3:**
Re-run test attached to PR #9771.
Verify that the screenshot taken via `toImage()` can still be viewed via `previewImage()`.

**Test 4:**
_(You need to take photos and videos using the Android device's camera before starting the below test procedure.)_

1. Build and run the below code on an Android 7.0 or newer device.
2. An alert is displayed upon app launch. Tap the "Photo" button.
3. Select a photo from the gallery.
4. Verify the Titanium app display the selected photo.
5. Back out of the app and re-launch it.
6. An alert is displayed upon app launch. Tap the "Video" button.
7. Select a video from the gallery.
8. Verify the Titanium app plays the selected video.
9. Back out of the app and re-launch it.
10. An alert is displayed upon app launch. Tap the "Both" button.
11. Select a video from the gallery.
12. Verify the Titanium app plays the selected video.
13. Repeat steps 2-12 on an Android 4.1 device.

```javascript
var window = Ti.UI.createWindow();
window.addEventListener("open", function(e) {
	var dialog = Ti.UI.createAlertDialog(
	{
		message: "Which media type do you want to open?",
		buttonNames: ["Photo", "Video", "Both"],
	});
	dialog.addEventListener("click", function(e) {
		var mediaTypes;
		if (e.index === 0) {
			mediaTypes = [Ti.Media.MEDIA_TYPE_PHOTO];
		} else if (e.index === 1) {
			mediaTypes = [Ti.Media.MEDIA_TYPE_VIDEO];
		} else if (e.index === 2) {
			mediaTypes = [Ti.Media.MEDIA_TYPE_PHOTO, Ti.Media.MEDIA_TYPE_VIDEO];
		} else {
			Ti.API.info("@@@ Alert was canceled.");
			return;
		}
		Ti.Media.openPhotoGallery(
		{
			allowMultiple: false,
			allowEditing: false,
			autohide: true,
			mediaTypes: mediaTypes,
			success: function(e) {
				Ti.API.info("@@@ e: " + JSON.stringify(e));
				Ti.API.info("@@@ e.media.file: " + e.media.file);
				Ti.API.info("@@@ e.media.nativePath: " + e.media.nativePath);
				switch (e.mediaType) {
					case Ti.Media.MEDIA_TYPE_PHOTO:
					case Ti.Media.MEDIA_TYPE_LIVEPHOTO:
						window.add(Ti.UI.createImageView(
						{
							image: e.media,
							width: Ti.UI.FILL,
							height: Ti.UI.FILL,
						}));
						break;
					case Ti.Media.MEDIA_TYPE_VIDEO:
						window.add(Ti.Media.createVideoPlayer(
						{
							url: e.media.nativePath,
							autoplay: true,
							mediaControlStyle: Ti.Media.VIDEO_CONTROL_DEFAULT,
							scalingMode: Ti.Media.VIDEO_SCALING_ASPECT_FIT,
							width: Ti.UI.FILL,
							height: Ti.UI.FILL,
						}));
						break;
					default:
						break;
				}
			},
			cancel: function() {
				Ti.API.info("@@@ Photo gallery selection canceled.");
			},
			error: function() {
				Ti.API.info("@@@ Photo gallery selection error.");
			},
		});
	});
	dialog.show();
});
window.open();
```
